contk/dataloader/multi_turn_dialog.py

```diff
@@ -294,3 +297,193 @@ def _load_data(self):
+		vocab_list.extend(left_vocab)
+
+		self.word2id = {w: i for i, w in enumerate(vocab_list)}
+
+		print("valid vocab list length = %d" % self.valid_vocab_len)
+		print("vocab list length = %d" % len(vocab_list))
+		return vocab_list, valid_vocab_set
+
+	def _load_multi_ref_data(self):
+		r'''Loading dataset, invoked by `SwitchboardCorpus._load_data`
+		'''
+		filename = '%s/switchboard_corpus_multi_ref.jsonl' % self._file_path
+		origin_data = self._read_file(filename, add_pre_turn=False)
+		test_data = self._convert2ids(origin_data)
+		test_data['candidate'] = []
+		with open(filename, "r") as data_file:
+			idx = 0
+			for line in data_file:
+				line = json.loads(line)
+				test_data['session'][idx].append(\
+					[self.go_id, self.word2id['<d>'], self.eos_id, self.eot_id])
+				utt2id = lambda utt: list(map(self._valid_word2id, utt[1].strip().split()))
+				test_data['candidate'].append(list(map(utt2id, line['responses'])))
+				idx += 1
+		return test_data
+
+	def _load_data(self):
+		r'''Loading dataset, invoked by `MultiTurnDialog.__init__`
+		'''
+		origin_data = {}
+		for key in self.key_name:
+			origin_data[key] = self._read_file('%s/switchboard_corpus_%s.jsonl' % (self._file_path, key))
+
+		vocab_list, valid_vocab_set = self._build_vocab(origin_data)
+
+		data = {}
+		data_size = {s: 0 for s in self.key_name}
+		for key in self.key_name:
+			data[key] = self._convert2ids(origin_data[key])
+			data_size[key] = len(data[key]['session'])
+
+			vocab = list(chain(*chain(*(origin_data[key]['session']))))
+			vocab_num = len(vocab)
+			oov_num = len(list(filter(lambda word: word not in self.word2id, vocab)))
+			invalid_vocab_num = len(list(filter(lambda word: \
+											word not in valid_vocab_set, vocab))) - oov_num
+			sent_lens = list(map(len, chain(*origin_data[key]['session'])))
+			cut_word_num = np.sum(np.maximum(np.array(sent_lens) - self._max_sen_length + 2, 0))
+			turn_lens = list(map(len, origin_data[key]['session']))
+			cut_sent_num = np.sum(np.maximum(np.array(turn_lens) - self._max_turn_length, 0))
+			print(("%s set. invalid rate: %f, unknown rate: %f, max sentence length before cut: %d, " + \
+				   "cut word rate: %f\n\tmax turn length before cut: %d, cut sentence rate: %f") % \
+				  (key, invalid_vocab_num / vocab_num, oov_num / vocab_num, max(sent_lens), \
+				   cut_word_num / vocab_num, max(turn_lens), cut_sent_num / np.sum(turn_lens)))
+		if os.path.exists('%s/switchboard_corpus_multi_ref.jsonl' % self._file_path):
+			self.key_name.append('multi_ref')
+			data['multi_ref'] = self._load_multi_ref_data()
+			data_size['multi_ref'] = len(data['multi_ref']['session'])
+		return vocab_list, len(valid_vocab_set), data, data_size
+
+	def get_batch(self, key, index):
+		'''Get a batch of specified `index`.
+
+		Arguments:
+			key (str): must be contained in `key_name`
+			index (list): a list of specified index
+
+		Returns:
+			(dict): A dict at least contains what is in the return of MultiTurnDialog.get_batch
```

Comment:
Don't use `at least`. List all the items besides what returned by MultiTurnDialog.get_batch